### PR TITLE
MaxGasPrice reads from TOML config; allow disabling price reporting on a lane

### PR DESCRIPTION
**Pull request by matYang**
## Motivation
Step 1 of Batch Gas Price reporting, allows for option of disabling gas and token price reporting on a lane, and make `MaxGasPrice` same per chain, as opposed to having it potentially different per lane.

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/ccipcommit/initializers.go [hunk 1]
```diff
@@ -87,13 +88,13 @@ func UnregisterCommitPluginLpFilters(ctx context.Context, lggr logger.Logger, jb
 	versionFinder := factory.NewEvmVersionFinder()
 	unregisterFuncs := []func() error{
 		func() error {
-			return factory.CloseCommitStoreReader(lggr, versionFinder, params.commitStoreAddress, params.destChain.Client(), params.destChain.LogPoller(), params.sourceChain.GasEstimator(), qopts...)
+			return factory.CloseCommitStoreReader(lggr, versionFinder, params.commitStoreAddress, params.destChain.Client(), params.destChain.LogPoller(), params.sourceChain.GasEstimator(), params.sourceChain.Config().EVM().GasEstimator().PriceMax().ToInt(), qopts...)
```
Comment (matYang) [inline core/services/ocr2/plugins/ccip/ccipcommit/initializers.go:92]: confirmed with BIX that we can get `PriceMax` this way, and not worry about `PriceMaxKey`, closed core PR that exposes this in estimator interface: https://github.com/smartcontractkit/chainlink/pull/12272/files

File: core/services/ocr2/plugins/ccip/ccipcommit/ocr2_test.go [hunk 2]
```diff
@@ -707,11 +778,44 @@ func TestCommitReportingPlugin_validateObservations(t *testing.T) {
...
+			f:                      2,
+			priceReportingDisabled: true,
+			expValidObs:            []ccip.CommitObservation{obEmpty, obEmpty},
+			expError:               true,
+		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			obs, err := validateObservations(ctx, logger.TestLogger(t), destTokens, tc.f, tc.commitObservations)
```
Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/ccipcommit/ocr2_test.go:714]: nit: can avoid this if you keep the function pure and just pass in the bool

File: core/services/ocr2/plugins/ccip/internal/ccipdata/factory/commit_store.go [hunk 3]
```diff
@@ -44,9 +46,11 @@ func initOrCloseCommitStoreReader(lggr logger.Logger, versionFinder VersionFinde
 		return nil, err
 	}
 
+	lggr.Infow("Initializing CommitStore Reader", "version", version.String(), "masGasPrice", maxGasPrice.String())
```
Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/factory/commit_store.go]: huh why doesn't maxGasPrice.String panic in the tests when nil?

Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/factory/commit_store.go]: typo - masGasPrice

File: docs/CHANGELOG_CCIP.md [hunk 4]
```diff
@@ -5,8 +5,14 @@ All notable changes to the CCIP project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+
 ## 1.5.0 - Unreleased
 
+### Changed
+- `CommitStore` offchain config format changed:
+  - Removed `MaxGasPrice` and `SourceMaxGasPrice` fields. MaxGasPrice will be read from PriceMax in chain TOML.
```
Comment (connorwstein) [inline docs/CHANGELOG_CCIP.md]: Maybe for further clarification - the fields MaxGasPrice/SourceMaxGasPrice will be ignored if specified but won't error (non-breaking change)

File: docs/CHANGELOG_CCIP.md [hunk 5]
```diff
@@ -5,8 +5,14 @@ All notable changes to the CCIP project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+
 ## 1.5.0 - Unreleased
```
Comment (connorwstein) [inline docs/CHANGELOG_CCIP.md:8]: Could potentially be released in 1.4.X right?

File: core/services/ocr2/plugins/ccip/internal/ccipdata/factory/commit_store.go [hunk 6]
```diff
@@ -21,16 +23,16 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
 )
 
-func NewCommitStoreReader(lggr logger.Logger, versionFinder VersionFinder, address cciptypes.Address, ec client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator, pgOpts ...pg.QOpt) (ccipdata.CommitStoreReader, error) {
-	return initOrCloseCommitStoreReader(lggr, versionFinder, address, ec, lp, estimator, false, pgOpts...)
+func NewCommitStoreReader(lggr logger.Logger, versionFinder VersionFinder, address cciptypes.Address, ec client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator, maxGasPrice *big.Int, pgOpts ...pg.QOpt) (ccipdata.CommitStoreReader, error) {
```
Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/factory/commit_store.go]: lets be super clear this is sourceMaxGasPrice

File: core/services/ocr2/plugins/ccip/internal/ccipdata/factory/offramp.go [hunk 7]
```diff
@@ -22,16 +24,16 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
 )
 
-func NewOffRampReader(lggr logger.Logger, versionFinder VersionFinder, addr cciptypes.Address, destClient client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator, registerFilters bool, pgOpts ...pg.QOpt) (ccipdata.OffRampReader, error) {
-	return initOrCloseOffRampReader(lggr, versionFinder, addr, destClient, lp, estimator, false, registerFilters, pgOpts...)
+func NewOffRampReader(lggr logger.Logger, versionFinder VersionFinder, addr cciptypes.Address, destClient client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator, maxGasPrice *big.Int, registerFilters bool, pgOpts ...pg.QOpt) (ccipdata.OffRampReader, error) {
```
Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/factory/offramp.go]: ditto here - let's be very clear this is destMaxGasPrice

File: core/services/ocr2/plugins/ccip/internal/ccipdata/factory/commit_store.go (hunk 3 above)
Comment (matYang) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/factory/commit_store.go]: Afaik it's by Golang's design, it has a nil check in `func (x *Int) Text(base int) string`
https://stackoverflow.com/questions/42238624/calling-a-method-on-a-nil-struct-pointer-doesnt-panic-why-not

File: docs/CHANGELOG_CCIP.md (hunk 5 above)
Comment (matYang) [reply docs/CHANGELOG_CCIP.md:8]: Last time Da mentioned 1.5 is slated for April, it's probably more realistic to target that. Given we are already at 1.4.5, cutting this in 1.5 makes sense too.

File: core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go [hunk 8]
```diff
@@ -615,7 +611,7 @@ func (o *OffRamp) RegisterFilters(qopts ...pg.QOpt) error {
 	return logpollerutil.RegisterLpFilters(o.lp, o.filters, qopts...)
 }
 
-func NewOffRamp(lggr logger.Logger, addr common.Address, ec client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator) (*OffRamp, error) {
+func NewOffRamp(lggr logger.Logger, addr common.Address, ec client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator, maxGasPrice *big.Int) (*OffRamp, error) {
```
Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go]: paranoid - can we name this param destMaxGasPrice too, ditto for 1.2 etc

File: core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/commit_store.go [hunk 9]
```diff
@@ -381,7 +380,7 @@ func (c *CommitStore) RegisterFilters(qopts ...pg.QOpt) error {
 	return logpollerutil.RegisterLpFilters(c.lp, c.filters, qopts...)
 }
 
-func NewCommitStore(lggr logger.Logger, addr common.Address, ec client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator) (*CommitStore, error) {
+func NewCommitStore(lggr logger.Logger, addr common.Address, ec client.Client, lp logpoller.LogPoller, estimator gas.EvmFeeEstimator, maxGasPrice *big.Int) (*CommitStore, error) {
```
Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/commit_store.go]: can we name this sourceMaxGasPrice too

**Review by connorwstein — APPROVED** [review]

**Review by AnieeG — APPROVED** [review]

**Pull request merged**